awslabs/aws-lambda-powertools-typescript
# feat: publish lib as Lambda Layer

## Description of your changes

* Integrate layer publication in CI/CD release process
* Publish doc on layer usage
* Add example on how to use it with CDK

### How to verify this change

Release process: https://github.com/awslabs/aws-lambda-powertools-typescript/actions/runs/2826141922


A public version is already available under this ARN: `arn:aws:lambda:{REGION}:094274105915:layer:AWSLambdaPowertoolsTypeScript:1` that was published during `1.0.2` release. So can be used in any deployment. An example of its usage is available in our CDK examples and be deploy simply:

```
cd examples/cdk
npm ci
npm run cdk deploy
```

Get to your AWS Console lambda , select the Lambda called `MyLayeredFunction` and check it's code and execution.

E2E tests has been added in the layer publisher as well.

### Related issues, RFCs

#826 

### PR status

***Is this ready for review?:*** NO  
***Is it a breaking change?:*** NO

## Checklist

- [x] Check tracking is in place for our layer
- [x] Add doc in SAM README on how to set the layer
- [ ] Add doc on release process with manual update doc for version bump

---

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
